realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,777 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.android;
+
+import android.content.ContentResolver;
+import android.database.CharArrayBuffer;
+import android.database.ContentObservable;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.CursorIndexOutOfBoundsException;
+import android.database.DataSetObservable;
+import android.database.DataSetObserver;
+import android.net.Uri;
+import android.os.Bundle;
+
+import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmResults;
+import io.realm.internal.ColumnType;
+import io.realm.internal.TableOrView;
+
+/**
+ * This class exposes {@link io.realm.RealmResults} as a cursor.
+ * <p>
+ * A RealmCursor has the same thread restrictions as RealmResults, so it is not possible to move a RealmCursor between
+ * threads.
+ * <p>
+ * A RealmCursor is currently limited to only being able to display data from a single RealmClass, i.e. it is not
+ * possible to follow links to other objects.
+ */
+public class RealmCursor implements Cursor {
+
+    private static final String DEFAULT_ID_COLUMN = "_id";
+
+    // Needed because constants are only available from API 11
+    private static final int FIELD_TYPE_NULL;
+    private static final int FIELD_TYPE_INTEGER;
+    private static final int FIELD_TYPE_FLOAT;
+    private static final int FIELD_TYPE_STRING;
+    private static final int FIELD_TYPE_BLOB;
+    static {
+        if (android.os.Build.VERSION.SDK_INT >= 11) {
```

Comment:
This branch can be removed.

Constant values are always inlined(JLS 13.1 The Form of a Binary (3)).
This means there will be no reference to Cursor.FIELD* in class file.